### PR TITLE
Sanitize post titles for Insights (and other endpoints)

### PR DESCRIPTION
**Pull request by astralbodies**
Fixes: #377 

Sanitizes the post titles coming back from the API when used in stats. This follows roughly the same method used in WPiOS. Added a unit test to ensure the fix was proper.

Sample return data from the API:
```
{
  "found": 16,
  "posts": [
    {
      "ID": 58,
      "date": "2016-01-21T14:46:42+00:00",
      "title": "This &#8220;has&#8221; special &amp; characters",
      "URL": "https://ardwptest1.wordpress.com/2016/01/21/this-has-special-characters/"
    }
  ]
}
```
And the fixed output:
simulator screen shot jan 21 2016 10 20 49 am

Needs Review: @aerych

### CODE REVIEW

File: WordPressCom-Stats-iOS/Services/WPStatsServiceRemote.m [hunk 1]
```diff
@@ -1127,7 +1133,7 @@ - (AFHTTPRequestOperation *)operationForAuthorsForDate:(NSDate *)date
             for (NSDictionary *post in posts) {
                 StatsItem *postItem = [StatsItem new];
                 postItem.itemID = [post numberForKey:@"id"];
-                postItem.label = [[post stringForKey:@"title"] stringByDecodingXMLCharacters];
+                postItem.label = [[[post stringForKey:@"title"] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]] stringByDecodingXMLCharacters];
```
Comment (aerych) [inline WordPressCom-Stats-iOS/Services/WPStatsServiceRemote.m]: I kinda want to make a helper method for these, passing in `[postDict stringForKey:@"title"]` and returning the formatted title. What do ya think?

**Comment by astralbodies** [comment]
Yeah I thought about that. I'll extract it out to a method in the remote service class at a minimum.

File: StatsDemo/StatsDemo/Images.xcassets/AppIcon.appiconset/Contents.json [hunk 2]
```diff
@@ -66,6 +66,11 @@
       "idiom" : "ipad",
       "filename" : "Icon-76@2x.png",
       "scale" : "2x"
+    },
+    {
+      "idiom" : "ipad",
+      "size" : "83.5x83.5",
+      "scale" : "2x"
```
Comment (aerych) [inline StatsDemo/StatsDemo/Images.xcassets/AppIcon.appiconset/Contents.json]: This is weird. No filename is listed.  Something being sneaky?

Comment (astralbodies) [inline StatsDemo/StatsDemo/Images.xcassets/AppIcon.appiconset/Contents.json]: Stupid Xcode. It's missing that icon file and it keeps regenerating that placeholder.

**Comment by astralbodies** [comment]
@aerych Ready for another review. I'm not happy with the `#import` hack for the Swift class but it's something I have to do for now until I have the time to figure out how to configure CocoaPods right :smile:

**Comment by aerych** [comment]
Works great and all tests pass. 
:shipit:

**Comment by astralbodies** [comment]
Really appreciate the review, @aerych!

**Pull request merged**